This patch is backported from git master:

commit 9c651933a451d31d78d322b115be07ae3d8dc888
Author: Julien Danjou <julien@danjou.info>
Date:   Mon Sep 28 14:41:54 2009 +0200

    Revert "awful.layout.suit.magnifier: simplify"

    This reverts commit f3c57faf1db4252409277a8989ac8f7e6662a732.

diff --git a/lib/awful/layout/suit/magnifier.lua.in b/lib/awful/layout/suit/magnifier.lua.in
index 3248299..2d4a526 100644
--- a/lib/awful/layout/suit/magnifier.lua.in
+++ b/lib/awful/layout/suit/magnifier.lua.in
@@ -19,6 +19,7 @@ local client = require("awful.client")
 module("awful.layout.suit.magnifier")
 
 function arrange(p)
+    -- Fullscreen?
     local area = p.workarea
     local cls = p.clients
     local focus = capi.client.focus
@@ -28,14 +29,18 @@ function arrange(p)
     -- Check that the focused window is on the right screen
     if focus and focus.screen ~= p.screen then focus = nil end
 
-    -- If we don't have a focused window on our screen,
-    -- and we have at least one client, take the first one.
     if not focus and #cls > 0 then
         focus = cls[1]
         fidx = 1
     end
 
-    -- Abort if clients to arrange are present.
+    -- If focused window is not tiled, take the first one which is tiled.
+    if client.floating.get(focus) then
+        focus = cls[1]
+        fidx = 1
+    end
+
+    -- Abort if no clients are present
     if not focus then return end
 
     local geometry = {}
